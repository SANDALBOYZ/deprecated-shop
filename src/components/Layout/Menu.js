// @flow
import React, { useContext } from 'react'
import styled from '@emotion/styled'
import { rhythm } from 'utils/typography'
// Components
import { StateContext } from 'components/StateProvider'
import { Link } from 'gatsby'
import SearchBar from './SearchBar'
// CSS
import { fadeInLeftRight } from 'styles/animations.css.js'
import { HEADER_HEIGHT } from './Header'

const MenuContainer = styled.div`
  display: ${({ isOpen }) => isOpen ? 'block' : 'none'};
  position: fixed;
  top: ${HEADER_HEIGHT};
  z-index: 900;
  background-color: white;
  width: 100%;
  height: calc(100vh - ${HEADER_HEIGHT});
  padding: ${rhythm(1)};
`

const MenuNav = styled.ul`
  margin: 0;
  padding: 0;
`

const MenuNavItem = styled.li`
  font-size: 1.3rem;
  margin: 0;
  margin-bottom: ${rhythm(0.5)};
  padding: 0;
  list-style-type: none;
  font-weight: bold;
  animation: ${fadeInLeftRight} 0.75s ease;
`

const Menu = () => {
  const [state] = useContext(StateContext)
  const { menuIsOpen } = state

  return (
    <MenuContainer isOpen={menuIsOpen}>
      <SearchBar />
      <MenuNav>
        <MenuNavItem>
          <Link to='/products'>Shop</Link>
        </MenuNavItem>
        <MenuNavItem>
          <Link to='/media'>Media</Link>
        </MenuNavItem>
        <MenuNavItem>
          <Link to='/about'>About</Link>
        </MenuNavItem>
        <MenuNavItem>
          <Link to='/help'>Help</Link>
        </MenuNavItem>
      </MenuNav>
    </MenuContainer>
  )
}

export default Menu
