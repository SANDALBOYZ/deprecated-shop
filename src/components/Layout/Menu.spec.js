/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import Menu from './Menu'

xdescribe('<Menu />', () => {
  it('renders', () => {
    const wrapper = shallow(<Menu />)

    expect(wrapper).toMatchSnapshot()
  })
})
