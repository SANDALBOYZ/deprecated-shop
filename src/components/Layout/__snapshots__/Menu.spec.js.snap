// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <SearchBar />,
        <ForwardRef(render)>
          <ForwardRef(render)>
            <GatsbyLink
              to="/"
            >
              Shop
            </GatsbyLink>
          </ForwardRef(render)>
          <ForwardRef(render)>
            <GatsbyLink
              to="/about"
            >
              About
            </GatsbyLink>
          </ForwardRef(render)>
          <ForwardRef(render)>
            <GatsbyLink
              to="/help"
            >
              Help
            </GatsbyLink>
          </ForwardRef(render)>
        </ForwardRef(render)>,
      ],
      "isOpen": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef(render)>
              <GatsbyLink
                to="/"
              >
                Shop
              </GatsbyLink>
            </ForwardRef(render)>,
            <ForwardRef(render)>
              <GatsbyLink
                to="/about"
              >
                About
              </GatsbyLink>
            </ForwardRef(render)>,
            <ForwardRef(render)>
              <GatsbyLink
                to="/help"
              >
                Help
              </GatsbyLink>
            </ForwardRef(render)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <GatsbyLink
                to="/"
              >
                Shop
              </GatsbyLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Shop",
                "to": "/",
              },
              "ref": null,
              "rendered": "Shop",
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "__emotion_base": "li",
              "__emotion_forwardProp": undefined,
              "__emotion_real": [Circular],
              "__emotion_styles": Array [
                "
  font-size: 1.3rem;
  margin: 0;
  margin-bottom: ",
                "0.75rem",
                ";
  padding: 0;
  list-style-type: none;
  font-weight: bold;
  animation: ",
                "fadeInLeftRight",
                " 0.75s ease;
",
              ],
              "defaultProps": undefined,
              "displayName": "Styled(li)",
              "render": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <GatsbyLink
                to="/about"
              >
                About
              </GatsbyLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "About",
                "to": "/about",
              },
              "ref": null,
              "rendered": "About",
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "__emotion_base": "li",
              "__emotion_forwardProp": undefined,
              "__emotion_real": [Circular],
              "__emotion_styles": Array [
                "
  font-size: 1.3rem;
  margin: 0;
  margin-bottom: ",
                "0.75rem",
                ";
  padding: 0;
  list-style-type: none;
  font-weight: bold;
  animation: ",
                "fadeInLeftRight",
                " 0.75s ease;
",
              ],
              "defaultProps": undefined,
              "displayName": "Styled(li)",
              "render": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <GatsbyLink
                to="/help"
              >
                Help
              </GatsbyLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Help",
                "to": "/help",
              },
              "ref": null,
              "rendered": "Help",
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "__emotion_base": "li",
              "__emotion_forwardProp": undefined,
              "__emotion_real": [Circular],
              "__emotion_styles": Array [
                "
  font-size: 1.3rem;
  margin: 0;
  margin-bottom: ",
                "0.75rem",
                ";
  padding: 0;
  list-style-type: none;
  font-weight: bold;
  animation: ",
                "fadeInLeftRight",
                " 0.75s ease;
",
              ],
              "defaultProps": undefined,
              "displayName": "Styled(li)",
              "render": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "ul",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            "
  margin: 0;
  padding: 0;
",
          ],
          "defaultProps": undefined,
          "displayName": "Styled(ul)",
          "render": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "__emotion_base": "div",
      "__emotion_forwardProp": undefined,
      "__emotion_real": [Circular],
      "__emotion_styles": Array [
        "
  display: ",
        [Function],
        ";
  position: fixed;
  top: ",
        "3.75rem",
        ";
  z-index: 900;
  background-color: white;
  width: 100%;
  height: calc(100vh - ",
        "3.75rem",
        ");
  padding: ",
        "1.5rem",
        ";
",
      ],
      "defaultProps": undefined,
      "displayName": "Styled(div)",
      "render": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <SearchBar />,
          <ForwardRef(render)>
            <ForwardRef(render)>
              <GatsbyLink
                to="/"
              >
                Shop
              </GatsbyLink>
            </ForwardRef(render)>
            <ForwardRef(render)>
              <GatsbyLink
                to="/about"
              >
                About
              </GatsbyLink>
            </ForwardRef(render)>
            <ForwardRef(render)>
              <GatsbyLink
                to="/help"
              >
                Help
              </GatsbyLink>
            </ForwardRef(render)>
          </ForwardRef(render)>,
        ],
        "isOpen": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef(render)>
                <GatsbyLink
                  to="/"
                >
                  Shop
                </GatsbyLink>
              </ForwardRef(render)>,
              <ForwardRef(render)>
                <GatsbyLink
                  to="/about"
                >
                  About
                </GatsbyLink>
              </ForwardRef(render)>,
              <ForwardRef(render)>
                <GatsbyLink
                  to="/help"
                >
                  Help
                </GatsbyLink>
              </ForwardRef(render)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <GatsbyLink
                  to="/"
                >
                  Shop
                </GatsbyLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Shop",
                  "to": "/",
                },
                "ref": null,
                "rendered": "Shop",
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "li",
                "__emotion_forwardProp": undefined,
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  "
  font-size: 1.3rem;
  margin: 0;
  margin-bottom: ",
                  "0.75rem",
                  ";
  padding: 0;
  list-style-type: none;
  font-weight: bold;
  animation: ",
                  "fadeInLeftRight",
                  " 0.75s ease;
",
                ],
                "defaultProps": undefined,
                "displayName": "Styled(li)",
                "render": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <GatsbyLink
                  to="/about"
                >
                  About
                </GatsbyLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "About",
                  "to": "/about",
                },
                "ref": null,
                "rendered": "About",
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "li",
                "__emotion_forwardProp": undefined,
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  "
  font-size: 1.3rem;
  margin: 0;
  margin-bottom: ",
                  "0.75rem",
                  ";
  padding: 0;
  list-style-type: none;
  font-weight: bold;
  animation: ",
                  "fadeInLeftRight",
                  " 0.75s ease;
",
                ],
                "defaultProps": undefined,
                "displayName": "Styled(li)",
                "render": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <GatsbyLink
                  to="/help"
                >
                  Help
                </GatsbyLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Help",
                  "to": "/help",
                },
                "ref": null,
                "rendered": "Help",
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "li",
                "__emotion_forwardProp": undefined,
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  "
  font-size: 1.3rem;
  margin: 0;
  margin-bottom: ",
                  "0.75rem",
                  ";
  padding: 0;
  list-style-type: none;
  font-weight: bold;
  animation: ",
                  "fadeInLeftRight",
                  " 0.75s ease;
",
                ],
                "defaultProps": undefined,
                "displayName": "Styled(li)",
                "render": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "ul",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              "
  margin: 0;
  padding: 0;
",
            ],
            "defaultProps": undefined,
            "displayName": "Styled(ul)",
            "render": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "__emotion_base": "div",
        "__emotion_forwardProp": undefined,
        "__emotion_real": [Circular],
        "__emotion_styles": Array [
          "
  display: ",
          [Function],
          ";
  position: fixed;
  top: ",
          "3.75rem",
          ";
  z-index: 900;
  background-color: white;
  width: 100%;
  height: calc(100vh - ",
          "3.75rem",
          ");
  padding: ",
          "1.5rem",
          ";
",
        ],
        "defaultProps": undefined,
        "displayName": "Styled(div)",
        "render": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
